### PR TITLE
Extract the user telemetry service so we can make it one package.

### PR DESCRIPTION
**Pull request by AWare**
## What does this change?
This extracts the telemetry client from prosemirror-typerighter so it can be used more generally. It should be a copy of the existing [Typerighter version of this code](https://github.com/guardian/prosemirror-typerighter/blob/main/src/ts/services/TelemetryService.ts).

## How can we measure success?
That when published, it can be used in prosemirror-typerighter and elsewhere.

## Testing

We should be able to test it using `yalc` or similar and linking the output to other projects.

### CODE REVIEW

File: projects/user-telemetry-client/src/ITelemetryData.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
+export interface ITelemetryEvent {
+  /**
+   * The application sending the event
+   */
+  app: string;
+
+  /**
+   * The application stage, e.g. 'CODE' | 'PROD'
+   */
+  stage: string;
```
Comment (jonathonherbert) [inline projects/user-telemetry-client/src/ITelemetryData.ts:10]: We can take this opportunity to use the interface already defined in this library in `definitions`. This definition supplants what's already there.

File: projects/user-telemetry-client/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,69 @@
+{
+  "compilerOptions": {
+    /* Visit https://aka.ms/tsconfig.json to read more about this file */
```
Comment (jonathonherbert) [inline projects/user-telemetry-client/tsconfig.json:3]: VSCode gives us excellent intellisense for compiler options – we can probably lose the comments.

File: projects/user-telemetry-client/yarn.lock [hunk 3]
```diff
@@ -0,0 +1,18 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
+# yarn lockfile v1
```
Comment (jonathonherbert) [inline projects/user-telemetry-client/yarn.lock:2]: We use npm for other projects in this repo – we should use npm here to be consistent, and if we'd like to convert this to yarn (which would be welcome!), that should be another PR.

File: projects/user-telemetry-client/tsconfig.json (hunk 2 above)
Comment (AWare) [reply projects/user-telemetry-client/tsconfig.json:3]: I said it was early :P

File: projects/user-telemetry-client/src/ITelemetryData.ts (hunk 1 above)
Comment (AWare) [reply projects/user-telemetry-client/src/ITelemetryData.ts:10]: This may be awkward to compile.

Comment (jonathonherbert) [reply projects/user-telemetry-client/src/ITelemetryData.ts:10]: Let's find out! If that proves to be the case, I'm sure we can figure out a way of making it less so.

File: projects/user-telemetry-client/yarn.lock (hunk 3 above)
Comment (AWare) [reply projects/user-telemetry-client/yarn.lock:2]: This will take some amount of teamcity retooling unfortunately, so I might. (Unless we go for manual publication)

**Review by jonathonherbert — APPROVED** [review]
Code looks good! Tested in Composer w/ `yalc`, performs as expected.

**Review by jonathonherbert — APPROVED** [review]
Code looks good! Tested in Composer w/ `yalc`, performs as expected.

**Review by jonathonherbert — APPROVED** [review]
Code looks good! Tested in Composer w/ `yalc`, performs as expected.
